pkg/segment/query/querystatus.go

```diff
@@ -169,6 +170,10 @@ func StartQuery(qid uint64, async bool, cleanupCallback func()) (*RunningQuerySt
 		return nil, fmt.Errorf("qid has already been started")
 	}
 
+	if len(allRunningQueries) == MAX_RUNNING_QUERIES {
```

Comment:
nit: this should also work, but typically in such situations I prefer to use `>` instead of exact comparison

Reply:
addressed in the latest commit.